luatest/server.lua

```diff
@@ -297,6 +412,11 @@ end
 function Server:start(opts)
     checks('table', {wait_until_ready = '?boolean'})
 
+    opts = opts or {}
+    if self.config_file and opts.wait_until_ready == nil then
+        opts.wait_until_ready = self.net_box_uri ~= nil
+    end
```

Comment:
What is the idea of this block?

Reply:
Wait until the server is ready when the config file is provided and netbox URI is not `nil`.

Reply:
OK, I left a comment to the actual place of this code: https://github.com/tarantool/luatest/pull/384#discussion_r1699936510.